numpy/_core/src/multiarray/item_selection.c

```diff
@@ -337,14 +337,16 @@ PyArray_TakeFrom(PyArrayObject *self0, PyObject *indices0, int axis,
         goto fail;
     }
 
-    Py_XDECREF(indices);
-    Py_XDECREF(self);
     if (out != NULL && out != obj) {
-        Py_INCREF(out);
-        PyArray_ResolveWritebackIfCopy(obj);
+        if (PyArray_ResolveWritebackIfCopy(obj) < 0) {
+            goto fail;
+        }
```

Comment:
This is an unrelated fix for a preexisting bug I ran into during debugging. The error value is discarded here and this function can error (it did in my testing) so we should have error checking.